site/src/sphinx/client-grpc.rst

```diff
@@ -157,6 +157,42 @@ As you might have noticed already, we decorated the client using :api:`LoggingCl
 requests and responses. You might be interested in decorating a client using other decorators, for example
 to gather metrics. Please also refer to :api:`ClientBuilder` for more configuration options.
 
+Exception propagation
+======================
+
+If you have enabled ``Flags.verboseResponses()`` in the server being accessed, then any exception during
+processing in the server will be returned to the client as a :api:`StatusCauseException` attached to the
+normal gRPC ``Status``. This can be used for programmatic access to the exception that happened in the
+server. In this example, the server always fails with ``throw new IllegalStateException("Failed!");``
+
+.. code-block:: java
+
+    import com.linecorp.armeria.client.Clients;
+    import com.linecorp.armeria.common.grpc.StatusCauseException;
+
+    import io.grpc.StatusRuntimeException;
+
+    HelloServiceBlockingStub helloService = Clients.newClient(
+            "gproto+http://127.0.0.1:8080/",
+            HelloServiceBlockingStub.class); // or HelloServiceFutureStub.class or HelloServiceStub.class
+
+    HelloRequest request = HelloRequest.newBuilder().setName("Armerian World").build();
+    try {
+        HelloReply reply = helloService.hello(request);
+    } catch (StatusRuntimeException e) {
+        if (e.getCause() instanceof StatusCauseException) {
+            StatusCauseException cause = (StatusCauseException) e.getCause();
+            // The name of the class of the exception in the server can be accessed.
+            assert cause.getClassName() == "java.lang.IllegalStateException";
+            // The exception's message is a combination of both this class name and original message.
+            assert cause.getMessage() == "java.lang.IllegalStateException: Failed!";
```

Comment:
- Could you also add an assert for `getOriginalMessage()`?
- `==` -> `.equals()`